soot-infoflow-android/src/soot/jimple/infoflow/android/data/AndroidField.java

```diff
@@ -0,0 +1,65 @@
+package soot.jimple.infoflow.android.data;
+
+import soot.SootField;
+import soot.jimple.infoflow.data.SootFieldAndMethod;
+import soot.jimple.infoflow.sourcesSinks.definitions.SourceSinkType;
+
+public class AndroidField extends SootFieldAndMethod {
```

Comment:
I think you don't need an Android-specific class here. Just move the source/sink type to the base class, since it makes sense for normal Java data flow analysis as well. For methods, we only had this distinction, because our ``AndroidMethod`` stores permissions, which are Android-specific. For fields, there is no Android-specific data.